File: scenarios/c-material-theme-lighter/input.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char *name;
    int age;
} Person;

int compare(const void *a, const void *b) {
    return ((Person *)a)->age - ((Person *)b)->age;
}

int main(void) {
    Person people[] = {{"Alice", 30}, {"Bob", 25}, {"Carol", 35}};
    size_t n = sizeof(people) / sizeof(people[0]);
    qsort(people, n, sizeof(Person), compare);
    for (size_t i = 0; i < n; i++) {
        printf("%s is %d\n", people[i].name, people[i].age);
    }
    return 0;
}

File: scenarios/go-laserwave/input.go
package main

import (
	"errors"
	"fmt"
	"sort"
)

type ByAge []Person

type Person struct {
	Name string
	Age  int
}

func (a ByAge) Len() int           { return len(a) }
func (a ByAge) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
func (a ByAge) Less(i, j int) bool { return a[i].Age < a[j].Age }

func oldest(people []Person) (Person, error) {
	if len(people) == 0 {
		return Person{}, errors.New("empty")
	}
	sort.Sort(ByAge(people))
	return people[len(people)-1], nil
}

func main() {
	people := []Person{
		{"Alice", 30},
		{"Bob", 25},
		{"Carol", 35},
	}
	p, err := oldest(people)
	if err != nil {
		panic(err)
	}
	fmt.Printf("oldest: %s (%d)\n", p.Name, p.Age)
}
